jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/execution/sam/JavaSamInvokeRunnerTest.kt

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.projectRoots.ProjectJdkTable
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.testFramework.IdeaTestUtil
+import com.intellij.testFramework.runInEdtAndGet
+import com.intellij.testFramework.runInEdtAndWait
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.aws.toolkits.core.lambda.LambdaRuntime
+import software.aws.toolkits.jetbrains.core.credentials.MockCredentialManagerRule
+import software.aws.toolkits.jetbrains.services.lambda.execution.local.LocalLambdaRunConfiguration
+import software.aws.toolkits.jetbrains.services.lambda.execution.local.createTemplateRunConfiguration
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamCommonTestUtils.addSamTemplate
+import software.aws.toolkits.jetbrains.utils.getState
+import software.aws.toolkits.jetbrains.utils.rules.HeavyJavaCodeInsightTestFixtureRule
+
+class JavaSamInvokeRunnerTest {
+    @Rule
+    @JvmField
+    val projectRule = HeavyJavaCodeInsightTestFixtureRule()
+
+    @Rule
+    @JvmField
+    val credentialManager = MockCredentialManagerRule()
+
+    private val mockCredsId = "mockId"
+
+    @Before
+    fun setUp() {
+        // force fixture to be created before write action in EDT
+        projectRule.fixture
+        credentialManager.addCredentials(mockCredsId)
+
+        val sdk = IdeaTestUtil.getMockJdk18()
+        runInEdtAndWait {
+            runWriteAction {
+                ProjectJdkTable.getInstance().addJdk(sdk, projectRule.fixture.projectDisposable)
+                ProjectRootManager.getInstance(projectRule.project).projectSdk = sdk
+            }
+        }
+    }
+
+    @Test
+    fun addsJavaHomeEnvVarForJavaTemplate() {
```

Comment:
Shouldn't there be a test for handler based in here as well?

Reply:
added